### PR TITLE
Insert .notdef glyph for missing glyphs

### LINKED ISSUE #65: improve handling of missing chars
mode=node simply drops missing chars, while xetex and harf-mode either insert spaces or the "missing char" glyph. I think the later is better and it should be done if possible by mode=node too. 

~~~~
\documentclass{article}
\begin{document}
x\SS y
\end{document}
~~~~

## output with lualatex (mode=node)

image

## output with harftex (+ \usepackage{harfload} or lua-code in the format) or xelatex

image

## output of a libertinus example with harftex

image

### PR DESCRIPTION
**Pull request by zauguin**
Fixes #65.

### CODE REVIEW

File: src/luaotfload-notdef.lua [hunk 1]
```diff
@@ -0,0 +1,67 @@
...
+      insert_after(cur, cur, notdefnode)
+      protect_glyph(cur)
+    end
+  end end
+end
+
+otffeatures.register {
+  name        = "notdef",
+  description = "Add notdef glyphs",
+  default     = 1,
```
Comment (zauguin) [inline src/luaotfload-notdef.lua]: Making this the default is actually a breaking change. So the more conservative choice would be
```suggestion
  default     = false,
```

**Comment by u-fischer** [comment]
I think we should go here for the breaking change. Getting a visual feedback for missing chars is useful and it is done by xelatex/harftex too, so it is more consistent. I have already merged it locally and will run the tests on it, (and document it).

File: src/luaotfload-notdef.lua [hunk 2]
```diff
@@ -0,0 +1,67 @@
...
+local getfont            = font.getfont
+local setfont            = node.direct.setfont
+local getwhd             = node.direct.getwhd
+local insert_after       = node.direct.insert_after
+local traverse_char      = node.direct.traverse_char
+local protect_glyph      = node.direct.protect_glyph
+local otffeatures        = fonts.constructors.newfeatures "otf"
+
+local function setnotdef(tfmdata, factor)
+  tfmdata.notdefcode = tfmdata.resources.unicodes[".notdef"]
```
Comment (khaledhosny) [inline src/luaotfload-notdef.lua]: I'm not sure relying on fonts having correct glyph names is a good idea. In OpenType glyph names are of secondary imporance and fonts can go away with all sorts of broken naming, so I'd not be surprised if there are fonts somewhere that names a regular glyph `.notdef`. Always using index 0 is probably a better idea for OpenType/TrueType fonts.

**Comment by u-fischer** [comment]
@zauguin with harftex + mode=node the newest code reports two missing chars:

~~~~
\font\test={file:lmroman10-regular.otf:mode=node;}\test
Latin Modern: x\SS y
~~~~

gives 

~~~~
{c:/texlive/2019/texmf-var/fonts/map/pdftex/updmap/pdftex.map}
Missing character: There is no ẞ (U+1E9E) in font file:lmroman10-regular.otf:m
ode=node;!
Missing character: There is no 󰀀 (U+F0000) in font file:lmroman10-regular.otf
:mode=node;!
] (./test-utf8.aux))
~~~~

**Pull request closed without merging**